addons/l10n_in/tests/test_invoice_hsn_summary.py

```diff
@@ -0,0 +1,618 @@
+
+    def assert_hsn_summary(self, invoice, expected_values):
+        hsn_summary = invoice._l10n_in_get_hsn_summary_table()
+        self.assertEqual(
+            {k: len(v) if k == 'items' else v for k, v in hsn_summary.items()},
+            {k: len(v) if k == 'items' else v for k, v in expected_values.items()},
+        )
+        self.assertEqual(len(hsn_summary['items']), len(expected_values['items']))
+        for item, expected_item in zip(hsn_summary['items'], expected_values['items']):
+            self.assertDictEqual(item, expected_item)
+
+    @freeze_time('2019-01-01')
+    def create_invoice(self, invoice_line_ids, **kwargs):
+        return self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            **kwargs,
+            'invoice_line_ids': invoice_line_ids,
+        })
+
+    def test_l10n_in_hsn_summary_1(self):
+        """ Test GST/IGST taxes. """
+        invoice = self.create_invoice(invoice_line_ids=[
+            Command.create({
+                'product_id': self.product_a.id,
+                'quantity': 2.0,
+                'price_unit': 100,
+                'tax_ids': [Command.set(self.gst_5.ids)], #Tax: 10
+            }),
+            Command.create({
+                'product_id': self.product_c.id,
+                'quantity': 1.0,
+                'price_unit': 600,
+                'tax_ids': [Command.set(self.gst_5.ids)], #Tax: 30
+            }),
+            Command.create({
+                'l10n_in_hsn_code': self.test_hsn_code_1,
+                'quantity': 5.0,
+                'price_unit': 300,
+                'product_uom_id': self.uom_unit.id,
+                'tax_ids': [Command.set(self.gst_5.ids)], #Tax: 75
+            }),
+
+            Command.create({
+                'product_id': self.product_a.id,
+                'quantity': 2.0,
+                'price_unit': 100,
+                'tax_ids': [Command.set(self.gst_18.ids)], #Tax: 36
+            }),
+            Command.create({
+                'product_id': self.product_c.id,
+                'quantity': 1.0,
+                'price_unit': 600,
+                'tax_ids': [Command.set(self.gst_18.ids)], #Tax: 108
+            }),
+            Command.create({
+                'l10n_in_hsn_code': self.test_hsn_code_1,
+                'quantity': 5.0,
+                'price_unit': 300,
+                'product_uom_id': self.uom_unit.id,
+                'tax_ids': [Command.set(self.gst_18.ids)], #Tax: 270
+            }),
+        ])
+
+        self.assertRecordValues(invoice, [{
+            'amount_untaxed': 4600.0,
+            'amount_tax': 529.0,
+            'amount_total': 5129.0,
+        }])
+
+        self.assert_hsn_summary(invoice, {
+            'has_igst': False,
+            'has_gst': True,
+            'has_cess': False,
+            'nb_columns': 7,
+            'display_uom': False,
```

Comment:
@jco-odoo It's due to we are checking `display_uom` statically, we need to check if `uom` is active or not in the test case before we proceed ahead.

May be I might have forgot this case while writing this test case

Reply:
Let me check if I can give it a quick fix